crates/engine/tree/src/tree.rs

```diff
@@ -1043,7 +1064,7 @@ where
             return None
         }
 
-        if self.is_backfill_active {
+        if self.is_backfill_active.is_active() {
```

Comment:
this needs a different check: `!is_idle` because we need to treat pending and active the same and are only interested in is idle or not idle

Reply:
fixed